Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -924,6 +942,31 @@ def prod {ι : Type u} (f : ι → Cardinal) : Cardinal :=
 theorem mk_pi {ι : Type u} (α : ι → Type v) : #(∀ i, α i) = prod fun i => #(α i) :=
   mk_congr <| Equiv.piCongrRight fun _ => outMkEquiv.symm
 
+theorem mk_pi_congr {ι ι' : Type u} {f : ι → Type v} {g : ι' → Type v} (e : ι ≃ ι')
+    (h : ∀ i, #(f i) = #(g (e i))) : #(Π i, f i) = #(Π i, g i) :=
+  mk_congr <| Equiv.piCongr e fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
+
+theorem mk_pi_congr_prop {ι ι' : Prop} {f : ι → Type v} {g : ι' → Type v} (e : ι ↔ ι')
+    (h : ∀ i, #(f i) = #(g (e.mp i))) : #(Π i, f i) = #(Π i, g i) :=
+  mk_congr <| Equiv.piCongr (.ofIff e) fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
+
+theorem mk_pi_congr' {ι : Type u} {ι' : Type v} {f : ι → Type max w (max u v)}
+    {g : ι' → Type max w (max u v)} (e : ι ≃ ι')
+    (h : ∀ i, #(f i) = #(g (e i))) : #(Π i, f i) = #(Π i, g i) :=
+  mk_congr <| Equiv.piCongr e fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
```

Comment:
This looks pretty hard to apply; perhaps it should be stated with `lift`s in the equalities?

Reply:
I just wanted `mk_pi_congr` and `mk_pi_congr'` to cover all the cases where the outputs are in the same universe. We can also make lift versions, but maybe in another PR? This already inflated quite a bit from the single sigma congr  lemma I wanted it for originally haha.

Anyways if `mk_pi_congr'` is too obscure I don't mind removing it.

Reply:
@vihdzp, I think you've worked with `lift` more than anyone; what's your preference here?

Reply:
The usual design pattern is to first prove a version with general universes using `lift`, and to then use that to prove a more convenient version where the universes match. In this case, the lift version would look like this:
```lean 4
theorem mk_pi_congr_lift {ι : Type u} {ι' : Type v} {f : ι → Type w} {g : ι' → Type x} 
    (e : ι ≃ ι') (h : ∀ i, lift.{x} #(f i) = lift.{w} #(g (e i))) :
    lift.{max v x} #(Π i, f i) = lift.{max u w} #(Π i, g i) :=
  Cardinal.lift_mk_eq'.2 ⟨Equiv.piCongr e fun i ↦ Classical.choice <| Cardinal.lift_mk_eq'.1 (h i)⟩
```

Reply:
Presumably that should be called `lift_mk_congr_...`?

Reply:
The convention seems to be to suffix `lift` lemmas like this. See e.g. [`Cardinal.mk_image_le_lift`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Cardinal/Basic.html#Cardinal.mk_image_le_lift), [`Cardinal.mk_range_eq_lift`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Cardinal/Basic.html#Cardinal.mk_range_eq_lift), [`Cardinal.mk_iUnion_le_lift`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/SetTheory/Cardinal/Basic.html#Cardinal.mk_iUnion_le_lift), etc.

Reply:
@vihdzp Thanks for the lift version! I added it along with a sigma version. I don't see a way to use it to prove the non lift versions more easily here so I kept them the same.